VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs

```diff
@@ -7,19 +8,16 @@ public static class SortInfoExtension
     {
         public static SortInfo[] EnsureIdIncluded(this SortInfo[] sortInfos)
         {
-            if (!sortInfos.IsNullOrEmpty())
+            var sortInfoList = sortInfos?.ToList() ?? new List<SortInfo>();
```

Comment:
Don't create new list if original sortInfos array already contains 'id'

Reply:
I'm create new list when original sortInfos == null only

Reply:
`ToList()` creates new list

Reply:
Probably Artem meant we could change the result under condition
`if (!sortInfoList?.Any(x => x.SortColumn.EqualsInvariant("id")))`

Reply:
I mean there is no need to create new list and then convert it back to array if we already have 'id' in original array.